tests/flytekit/unit/core/test_interface.py

```diff
@@ -213,6 +213,18 @@ def z(
     assert not params.parameters["c"].required
     assert params.parameters["c"].default.scalar.none_type == Void()
 
+    def z(a: int | None = None, b: str | None = None, c: typing.List[int] | None = None) -> typing.Tuple[int, str]:
```

Comment:
Should we skip this test as well?

Reply:
we should split this portion of the `test_parameters_and_defaults` test into its own test.

It's not failing in CI now because we only [run](https://github.com/flyteorg/flytekit/blob/master/.github/workflows/pythonbuild.yml#L28-L32) on python 3.12 in PRs. This test would start to fail in the nightly runs.